llvm/lib/Target/AIE/AIE2Subtarget.cpp

```diff
@@ -76,3 +76,46 @@ const RegisterBankInfo *AIE2Subtarget::getRegBankInfo() const {
 InstructionSelector *AIE2Subtarget::getInstructionSelector() const {
   return InstSelector.get();
 }
+
+std::bitset<32>
+AIE2Subtarget::getMemoryBanksFromAddressSpace(unsigned AddrSpace) const {
```

Comment:
Maybe we could move that to AIE2InstrInfo instead? Then we can make that a virtual in `AIEBaseInstrInfo`, which is a real base class for all AIE `InstrInfo` classes.

Reply:
As discussed earlier we would like this to use with AA pass based on AddressSpace, and thus decided to keep it here moving there can make things challenging.